src/js/overlay.js

```diff
@@ -225,16 +219,33 @@ Overlay.prototype.show = function() {
 
 	this.closeOnExternalClick = this.closeOnExternalClick.bind(this);
 	this.delegates.doc.on('click', 'body', this.closeOnExternalClick);
+	this.delegates.doc.on('touchstart', 'body', this.closeOnExternalClick);
```

Comment:
Why do you think it's naive? It's like detecting click but in the case of
touch, that are different touch ones, and if you only want to listen to
touch like it was a click, touchstart seems to be the one most people use

-- 

------------------------------
This email was sent by a company owned by Pearson plc, registered office at 
80 Strand, London WC2R 0RL.  Registered in England and Wales with company 
number 53723.